inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_nms_5_to_legacy.cpp

```diff
@@ -7,49 +7,54 @@
-#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/opsets/opset5.hpp>
 #include <legacy/ngraph_ops/nms_ie.hpp>
 #include <ngraph/rt_info.hpp>
 #include <transformations/utils/utils.hpp>
 
-#include "legacy/transformations/convert_opset1_to_legacy/convert_nms_4_to_legacy.hpp"
+#include "legacy/transformations/convert_opset1_to_legacy/convert_nms_5_to_legacy.hpp"
 
-NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertNMS4ToLegacyMatcher, "ConvertNMS4ToLegacyMatcher", 0);
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertNMS5ToLegacyMatcher, "ConvertNMS5ToLegacyMatcher", 0);
 
-ngraph::pass::ConvertNMS4ToLegacyMatcher::ConvertNMS4ToLegacyMatcher() {
+ngraph::pass::ConvertNMS5ToLegacyMatcher::ConvertNMS5ToLegacyMatcher() {
     auto boxes = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1000, 4});
     auto scores = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1000});
-    auto max_output_boxes_per_class = ngraph::opset4::Constant::create(element::i64, Shape{}, {10});
-    auto iou_threshold = ngraph::opset4::Constant::create(element::f32, Shape{}, {0.75});
-    auto score_threshold = ngraph::opset4::Constant::create(element::f32, Shape{}, {0.7});
-    auto nms = std::make_shared<ngraph::opset4::NonMaxSuppression>(boxes, scores, max_output_boxes_per_class,
-                                                                   iou_threshold, score_threshold);
+    auto max_output_boxes_per_class = ngraph::opset5::Constant::create(element::i64, Shape{}, {10});
+    auto iou_threshold = ngraph::opset5::Constant::create(element::f32, Shape{}, {0.75});
+    auto score_threshold = ngraph::opset5::Constant::create(element::f32, Shape{}, {0.7});
+    auto soft_nms_sigma = ngraph::opset5::Constant::create(element::f32, Shape{}, {0.25});
+    auto nms = std::make_shared<ngraph::opset5::NonMaxSuppression>(boxes, scores, max_output_boxes_per_class,
+                                                                   iou_threshold, score_threshold, soft_nms_sigma);
 
     ngraph::matcher_pass_callback callback = [](pattern::Matcher &m) {
-        auto nms_4 = std::dynamic_pointer_cast<ngraph::opset4::NonMaxSuppression>(m.get_match_root());
-        if (!nms_4) {
+        auto nms_5 = std::dynamic_pointer_cast<ngraph::opset5::NonMaxSuppression>(m.get_match_root());
+        if (!nms_5) {
             return false;
         }
 
-        const auto new_args = nms_4->input_values();
-        const auto& arg2 = new_args.size() > 2 ? new_args.at(2) : ngraph::opset4::Constant::create(element::i32, Shape{}, {0});
-        const auto& arg3 = new_args.size() > 3 ? new_args.at(3) : ngraph::opset4::Constant::create(element::f32, Shape{}, {.0f});
-        const auto& arg4 = new_args.size() > 4 ? new_args.at(4) : ngraph::opset4::Constant::create(element::f32, Shape{}, {.0f});
+        const auto new_args = nms_5->input_values();
+        const auto& arg2 = new_args.size() > 2 ? new_args.at(2) : ngraph::opset5::Constant::create(element::i32, Shape{}, {0});
```

Comment:
Why do we need such checks if you always create a fake inputs?

BTW:
I am ok with these checks and I think we shouldn't create fake inputs in the constructor of op, it looks like a hack.
@lazarevevgeny Do you agree?

Reply:
I don't remember the reason why this fake inputs are added. Let's try to remove them and see what will happen. We also should add some IE IR Reader tests to test the behaviour when some inputs are missing.

Reply:
Now NMS-5 at in input of the transformation can have any number of inputs, from 2 to 6. Hence, we need such checks. Fake nodes are created because `NonMaxSuppressionIE3` has only constructor with 6 input nodes. Fake inputs for previous NMS versions were created because corresponding legacy layers have only constructors with 5 input nodes.